### PR TITLE
Formalize null handling of type tokens

### PR DESCRIPTION
**Pull request by nadiawoodninja**
https://github.com/google/gson/issues/1831

### CODE REVIEW

File: gson/src/test/java/com/google/gson/GsonTest.java [hunk 1]
```diff
@@ -82,4 +86,16 @@ public void testClonedTypeAdapterFactoryListsAreIndependent() {
...
   }
+
+  public void testNullAdapterLookup() {
+    // Formalize the handling of null type tokens (previously it was partially handled, but still throwing an error)
+    // Note: ExpectedException cannot be used due to test suite extending TestCase
+    try {
+      new Gson().getAdapter((TypeToken<Object>) null);
+      assertTrue("Unreachable", false);
+    }
+    catch (NullPointerException e) {
```
Comment (eamonnmcmanus) [inline gson/src/test/java/com/google/gson/GsonTest.java:97]: Can you join this to the previous line?

```
} catch (NullPointerException e) {
```

File: gson/src/test/java/com/google/gson/GsonTest.java [hunk 2]
```diff
@@ -82,4 +86,16 @@ public void testClonedTypeAdapterFactoryListsAreIndependent() {
     }
     @Override public Object read(JsonReader in) throws IOException { return null; }
   }
+
+  public void testNullAdapterLookup() {
+    // Formalize the handling of null type tokens (previously it was partially handled, but still throwing an error)
+    // Note: ExpectedException cannot be used due to test suite extending TestCase
+    try {
+      new Gson().getAdapter((TypeToken<Object>) null);
+      assertTrue("Unreachable", false);
```
Comment (eamonnmcmanus) [inline gson/src/test/java/com/google/gson/GsonTest.java:95]: Just `fail()` would be fine here.

**Review by eamonnmcmanus — COMMENTED** [review]
Thanks!

File: gson/src/test/java/com/google/gson/GsonTest.java [hunk 3]
```diff
@@ -26,6 +27,9 @@
 import java.util.ArrayList;
 import java.util.HashMap;
 import junit.framework.TestCase;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
```
Comment (Marcono1234) [inline gson/src/test/java/com/google/gson/GsonTest.java:32]: These imports seem to be unused.

File: gson/src/test/java/com/google/gson/GsonTest.java [hunk 4]
```diff
@@ -82,4 +86,16 @@ public void testClonedTypeAdapterFactoryListsAreIndependent() {
     }
     @Override public Object read(JsonReader in) throws IOException { return null; }
   }
+
+  public void testNullAdapterLookup() {
+    // Formalize the handling of null type tokens (previously it was partially handled, but still throwing an error)
```
Comment (Marcono1234) [inline gson/src/test/java/com/google/gson/GsonTest.java:91]: Not sure if this comment line here is useful in the test code. To me this message only seems to be relevant regarding why this pull request or commit was created, but not why this test exists.

---

Note that unlike eamonnmcmanus, I am not a member of this project. Feel free to consider my comments only as suggestion.

**Comment by Marcono1234** [comment]
I am closing this pull request because this issue has been fixed as part of #2153. Thanks nonetheless for your pull request!

**Pull request closed without merging**